lib/bulk_insert/worker.rb

```diff
@@ -76,5 +91,22 @@ def save!
 
       self
     end
+
+    private
+
+    def database_type
+      @connection.adapter_name.underscore.to_sym
+    end
+
+    def sqlite3_strategy(rows)
+      sql = "INSERT INTO #{@table_name} (#{@column_names}) VALUES "
+      sql << rows.collect{|row| "(#{row})"}.join(',')
+    end
+
+    def oracle_strategy(rows)
```

Comment:
My concern about having this as a method on the worker itself is that we're opening a can of worms. Every adapter that needs special insert syntax now has to add a new method to to the worker in order to get that. I'd like to see this abstracted more like how ActiveRecord does it, with a module representing the specialized functionality. Ideally, instantiating a worker would then look at whichever database adapter is active on the `@connection`, and would check to see if there is a bulk-insert module for that adapter, pulling it in and extending the worker as needed. (Does that make sense?) In this way, it becomes a lot more "plug and play", much easier to extend, with fewer special cases.